syntax = "proto3";
package pool;

import "common.proto";


service Deployment {
    rpc produce_plan(ProducePlanRequest) returns (ProducePlanResponse);
    rpc deploy_plan(DeploymentRequest) returns (DeploymentResponse);
}


message ProducePlanRequest {
    repeated common.ModelId models_ids = 1 ;
    float latency_weight = 2;
    float energy_weight = 3;
    float device_max_energy = 4 ;
    repeated int32 requests_number = 5 ;
    repeated float max_noises = 6 ;
    string start_server = 7;
}

message ProducePlanResponse {
    string optimized_plan = 1 ;
}

message DeploymentRequest {
    string optimized_plan = 1;
}

message DeploymentResponse {
}
